hub/src/server/http.js

```diff
@@ -78,10 +81,11 @@ export function makeHttpServer(config: Object) {
 
   app.get('/hub_info/', (req: express.request,
                          res: express.response) => {
-    const challengeText = StorageAuthentication.challengeText(server.serverName)
+    const challengeText = getChallengeText(server.serverName)
     const readURLPrefix = driver.getReadURLPrefix()
-    writeResponse(res, { 'challenge_text' : challengeText,
-                         'read_url_prefix' : readURLPrefix }, 200)
+    writeResponse(res, { 'challenge_text': challengeText,
```

Comment:
We should consider not replying anything if the challenge text is too short (i.e. give back a 400).

Reply:
Hmm -- I like that approach, will add.